CRM/Core/ManagedEntities.php

```diff
@@ -600,16 +479,17 @@ protected function isModuleDisabled(string $module): bool {
 
   /**
    * @param array $declarations
-   *
-   * @return array
+   * @param string $moduleName
+   *   Filter to a single module.
    */
-  protected function cleanDeclarations(array $declarations): array {
-    foreach ($declarations as $name => &$declare) {
-      if (!array_key_exists('name', $declare)) {
-        $declare['name'] = $name;
+  protected function setDeclarations(array $declarations, $moduleName = NULL) {
```

Comment:
(*Non-blocker/General observation*) There's a tension around the lifecycle of `$declarations`, which has behavioral and stylistic dimensions:

* (Behaviorally) Are `$declarations` ephemeral things (_loaded+destroyed each time you use `ManagedEntities`_) or are they retained for a while (_re-used in multiple calls to `ManagedEntities`_)? How long should they live in memory?
* (Stylistically) Do we prefer to use more stateless design (eg `cleanupDeclarations(array):array`) or more stateful design (eg `setDeclarations(array,string): void`)?

A couple thoughtlets that stand out to me:

* There are two `public` operations relying on this data (`reconcile()` + `revert()`). Both of them rebuild the `$declarations` before doing their work. This behavior seems reasonable to me (*though maybe that belief merits inspection*). In any case, it appears consistent before+after patch.
* AFAICS, the `$moduleName` is never actually passed in to `setDeclarations()`. I was concerned it would mean some subtle corruption of `$this->declarations`. But it doesn't matter - b/c `reconcile()`+`revert()` will rebuild `$this->declarations` whenever they're called.
* It seems to me that (following the other changes) there's now less need to store data in `$this`. In the older design, it retained `$this->declarations` so that each `reconcileFooModules()` could do a pass through the list -- in the newer design, the planner does a single pass over `$declarations`.

Anyway, this comment has as many questions as answers, and these don't have to be resolved now. The relevant methods are `protected function`s in a _de-facto_ `final class`, so we can continue to revise based on other analysis or discussion.